python-avd/pyavd/_eos_designs/structured_config/underlay/prefix_lists.py

```diff
@@ -21,66 +22,55 @@ class PrefixListsMixin(Protocol):
-            sequence_numbers.extend(
-                {"sequence": index * 10, "action": f"permit {network} eq 32"}
-                for index, network in enumerate(
-                    collapse_addresses(get_ipv4_networks_from_pool(self.shared_utils.vtep_loopback_ipv4_pool)), start=len(sequence_numbers) + 1
-                )
-            )
+            for index, network in enumerate(
+                collapse_addresses(get_ipv4_networks_from_pool(self.shared_utils.vtep_loopback_ipv4_pool)), start=len(sequence_numbers) + 1
+            ):
+                sequence_numbers.append_new(sequence=index * 10, action=f"permit {network} eq 32")
 
         if self.inputs.vtep_vvtep_ip is not None and self.shared_utils.network_services_l3 is True and not self.shared_utils.is_wan_router:
             sequence_number = (len(sequence_numbers) + 1) * 10
-            sequence_numbers.append({"sequence": sequence_number, "action": f"permit {self.inputs.vtep_vvtep_ip}"})
-
-        prefix_lists = [{"name": "PL-LOOPBACKS-EVPN-OVERLAY", "sequence_numbers": sequence_numbers}]
+            sequence_numbers.append_new(sequence=sequence_number, action=f"permit {self.inputs.vtep_vvtep_ip}")
+        self.structured_config.prefix_lists.append_new(name="PL-LOOPBACKS-EVPN-OVERLAY", sequence_numbers=sequence_numbers)
 
         if self.shared_utils.underlay_multicast_rp_interfaces is not None:
-            sequence_numbers = [
-                {"sequence": (index + 1) * 10, "action": f"permit {interface.ip_address}"}
-                for index, interface in enumerate(self.shared_utils.underlay_multicast_rp_interfaces)
-            ]
-            prefix_lists.append({"name": "PL-LOOPBACKS-PIM-RP", "sequence_numbers": sequence_numbers})
+            sequence_numbers = EosCliConfigGen.PrefixListsItem.SequenceNumbers()
+            for index, interface in enumerate(self.shared_utils.underlay_multicast_rp_interfaces):
+                sequence_numbers.append_new(sequence=(index + 1) * 10, action=f"permit {interface.ip_address}")
+            self.structured_config.prefix_lists.append_new(name="PL-LOOPBACKS-PIM-RP", sequence_numbers=sequence_numbers)
 
         # For now only configure it with eBGP towards LAN.
         if self.shared_utils.wan_ha and self.shared_utils.use_uplinks_for_wan_ha and self.shared_utils.underlay_routing_protocol == "ebgp":
-            sequence_numbers = [
-                {"sequence": 10 * (index + 1), "action": f"permit {ipaddress.ip_network(ip_address, strict=False)}"}
-                for index, ip_address in enumerate(self.shared_utils.wan_ha_ip_addresses)
-            ]
-
+            sequence_numbers = EosCliConfigGen.PrefixListsItem.SequenceNumbers()
+            for index, ip_address in enumerate(self.shared_utils.wan_ha_ip_addresses):
+                sequence_numbers.append_new(sequence=10 * (index + 1), action=f"permit {ipaddress.ip_network(ip_address, strict=False)}")
             if sequence_numbers:
```

Comment:
just change the logic and move this check before geneerating the sequence_numbers and same for next one

```
if self.shared_utils.wan_ha_ip_addresses:
  < create sequence_numbers and add PL>
if self.shared_utils.wan_ha_peer_ip_addresses:
  < create sequence_numbers and add PL>
```

Reply:
Fixed!